prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -139,51 +141,72 @@ def get_output_mapping(self):
         }
 
     @staticmethod
-    def setup_session(kubeconfig_file, input_context) -> KubernetesSession:
+    def setup_session(
+        kubeconfig_file: str = None,
+        kubeconfig_content: dict = {},
+        input_context: str = None,
+    ) -> KubernetesSession:
         """
         Sets up the Kubernetes session.
 
         Args:
             kubeconfig_file (str): Path to the kubeconfig file.
+            kubeconfig_content (dict): Content of the kubeconfig file.
             input_context (str): Context name.
 
         Returns:
             Tuple: A tuple containing the API client and the context.
         """
         logger.info(f"Using kubeconfig file: {kubeconfig_file}")
         try:
-            config.load_kube_config(
-                config_file=(
-                    os.path.abspath(kubeconfig_file)
-                    if kubeconfig_file != "~/.kube/config"
-                    else os.path.expanduser(kubeconfig_file)
-                ),
-                context=input_context,
-            )
-        except ConfigException:
-            # If the kubeconfig file is not found, try to use the in-cluster config
-            logger.info("Using in-cluster config")
-            config.load_incluster_config()
-            context = {
-                "name": "In-Cluster",
-                "context": {
-                    "cluster": "in-cluster",  # Placeholder, as the real cluster name is not available
-                    "user": "service-account-name",  # Also a placeholder
-                },
-            }
-        else:
+            if kubeconfig_content:
+                config.load_kube_config_from_dict(
+                    kubeconfig_content, context=input_context
+                )
+            else:
+                try:
```

Comment:
Why do you need the double `try/except`?

Reply:
We were not handling errors in this method.